site-src/geps/gep-1058.md

```diff
@@ -0,0 +1,623 @@
+      reason: HostnameMismatch
+      message: The route could not be attached because of a hostname mismatch in a child route.
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: HTTPRoute
+metadata:
+  name: http-child
+spec:
+  parentRefs:
+  - name: gateway-a
+    sectionName: http
+  hostnames:
+    - testdomain.example.com
+  rules:
+    - backendRefs:
+      - name: foo-svc
+        port: 8080
+
+```
+
+> youngnick note: Uh-oh, this example is valid according to what I see as the rules, but there's no way to represent it without reworking the status completely.
+>
+> This will need a child route to have a parentRefs namespaced by its *inclusion path* rather than by its direct parent. I'm going to need to think about this some more.
+
+In this example, the HTTPRoute `http-parent` is able to attach to either Gateway by itself, but once the child is included, `gateway-a` will accept it, and `gateway-b` will not, since the hostnames don't match. This is a slightly contrived example, but serves to illustrate that we need a way for both `http-parent` and `http-child` to indicate that something has gone wrong.
+
+
+
+
+
+
+## Alternatives
+
+### Extending backendRef to allow targeting a Route
+
+This idea was to allow `backendRef` to, optionally, specify a route instead of a
+service. Some of the reasons for not taking this approach include:
+
+- It would require changes to `backendRefs`. No changes to `allowedRoutes` are
+  needed to use it for this. 
+- To minimize the changes to `backendRef`, it would require the owner of the
+  parent route to know the exact name of the child route. If the two routes are
+  in different namespaces, that might require cross-organization coordination
+  between administrators.
+- It would result in there being one way to attach routes to listeners and a
+  different way to attach routes to routes.
+
+## References
+
+The first draft of this GEP was presented in the document [K8s Gateway API -
+Route inclusion and delegation][first-draft]. It includes many comments from the
+Gateway API community.
+
+There are multiple other documents, GitHub Issues and discussions that are
+directly related to this topic and influenced the design of this GEP. This
+include the following:
+
+- [Request Filtering Between Gateways and Namespaced Routes · Issue #634 · kubernetes-sigs/gateway-api (github.com)](https://github.com/kubernetes-sigs/gateway-api/issues/634)
+- [Request Filtering Between Gateways and Namespaced Routes - Google Docs](https://docs.google.com/document/d/1-0mgRRAY784OgGQ1_LCOshpLLbeAtIr4eXd0YVYK4RY/edit#heading=h.8cfxzle5tmqb)
+- [Clarify how RouteGateways would work if we supported Route->Route delegation · Issue #411 · kubernetes-sigs/gateway-api (github.com)](https://github.com/kubernetes-sigs/gateway-api/issues/411)
+- [Route inclusion/delegation security and policy implications · Issue #1042 · kubernetes-sigs/gateway-api (github.com)](https://github.com/kubernetes-sigs/gateway-api/issues/1042)
```

Comment:
I think we likely need some more coverage in this GEP about how policy attachment interacts with route inclusion. Unfortunately I'm not sure what the answer is yet, but it does add another dimension of complexity here.

Reply:
Yes, I think that there's absolutely an action here to do some definition about how policy attachment interacts with route inclusion.

Reply:
This will be relevant for GAMMA too, e.g. thinking about how a service owner defining timeout policy on a mesh HTTPRoute for a service interacts with timeout defaults or override set by a gateway administrator. I think it might actually be relatively straightforward, just an additional level deep but with the same default/override pattern as already established.